src/mi/backtrace.c

```diff
@@ -69,12 +70,52 @@ unw_backtrace (void **buffer, int size)
   if (unlikely (tdep_trace (&cursor, buffer, &n) < 0))
     {
       unw_getcontext (&uc);
-      return slow_backtrace (buffer, size, &uc);
+      return slow_backtrace (buffer, size, &uc, 0);
     }
 
   return n;
 }
 
+int
+unw_backtrace2 (void **buffer, int size, unw_context_t* uc2)
+{
+  if (size == 0)
+    return 0;
+    
+  if (uc2 == NULL)
+    return unw_backtrace(buffer, size);
+
+  unw_cursor_t cursor;
+  // need to copy, because the context will be modified by tdep_trace
```

Comment:
uc looks like a shallow copy of uc2. Based on the comment, does this need to be a deep copy?